src/BenchmarkDotNet/ListBenchmarks/BenchmarkCasesPrinter.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections.Generic;
+
+namespace BenchmarkDotNet.ListBenchmarks
+{
+    internal class BenchmarkCasesPrinter : IBenchmarkCasesPrinter
+    {
+        private readonly IBenchmarkCasesPrinter printer;
+        
+        public BenchmarkCasesPrinter(ListBenchmarkCaseMode listBenchmarkCaseMode)
+        {
```

Comment:
this method could be simplified to 

```
=> listBenchmarkCaseMode ==  ListBenchmarkCaseMode.Tree
        ? new TreeBenchmarkCasesPrinter()
        :  new FlatBenchmarkCasesPrinter();
```

or if we want to keep the switch we should have 3 cases: `Tree`, `Flat` and `Disabled` (throw)